plugins/externalwrench/include/gazebo/ExternalWrench.hh

```diff
@@ -0,0 +1,83 @@
+#ifndef EXTERNALWRENCH_H
+#define EXTERNALWRENCH_H
+
+#include <iostream>
+#include <stdlib.h>
+#include <ctime>
+#include <stdio.h>
+#include <string>
+#include <memory>
+
+#include "gazebo/gazebo.hh"
+#include <gazebo/physics/PhysicsEngine.hh>
+#include <gazebo/common/Events.hh>
+#include <gazebo/common/Plugin.hh>
+#include <gazebo/transport/Node.hh>
+#include <gazebo/physics/Link.hh>
+#include <gazebo/physics/World.hh>
+#include <gazebo/physics/Model.hh>
+#include <GazeboYarpPlugins/common.h>
+
+#include <yarp/os/Network.h>
+#include <yarp/os/RpcServer.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/sig/Vector.h>
+#include <yarp/os/Thread.h>
+#include <yarp/os/Time.h>
+#include <yarp/os/Vocab.h>
+#include <yarp/os/Log.h>
+#include <yarp/os/LogStream.h>
+#include <yarp/math/Math.h>
+
+#include <boost/lexical_cast.hpp>
+
+using namespace gazebo;
+
+class ExternalWrench
+{
+private:
+
+   static int count;
```

Comment:
Everything seems to be ok from what I understand, but static variables are always a bit tricky do deal with. 
I don't think this needs to block the PR, but I think you can obtain something similar by setting the `count` to the in ExternalWrench constructor as you build it in the plugin class.

Reply:
@traversaro I am not following you exactly. The `count` variable is actually increased in the constructor of `ExternalWrench` https://github.com/robotology/gazebo-yarp-plugins/blob/f796908d68c22d046586b4248a9fc6bf79b1190d/plugins/externalwrench/src/ExternalWrench.cc#L22

Reply:
Yes, but what happens if there are two different externalwrench plugins in the same model? As far as I was able to see, everything should work fine, but keeping `count` as a static variable may lead to some subtle bugs if this code is modified in the future.

Reply:
>  what happens if there are two different externalwrench plugins in the same model? 

At the moment this is not possible to do by a user as there will be port address conflict while trying to open the rpc port with the model name. So, I believe the usage of `count` as static is good in this case.

Reply:
And what about two  different externalwrench plugins in two different model that are spawned together:  the count would be shared also in that casel

Reply:
While writing  https://github.com/robotology/gazebo-yarp-plugins/pull/418/files#r273692453 , I am a bit confused by  the `count` logic  even in the case of the single plugin. Let's say that three wrenches are added,  and they get the count `1`, `2` and `3` , where `1` and `3` have a duration of 100 seconds, and `2` instead has a duration of 1 seconds. After the the wrench 2 is deleted, the count counter is decremented to 2. If (let's say after 10 seconds) a new wrench is added, the `count` counter is increased again, and so the new wrench gets a `count` of 3, that is a index that is already used  by the wrench  `3`, and the visual representation will probably conflict!

Reply:
The conflict will happen for the visuals. So, it is better to not decrease the count of the wrenches after the duration is done. It is better to keep an increasing `wrenchCount` variable in `ApplyExternalWrench` and set the `wrenchIndex` variable of `ExternalWrench` class. This way each new wrench will have a different color without conflicts.